### ISSUE TITLE
Including cpuinfo_arm.h from an Apple M1 target.

### ISSUE THREAD

**Issue by Gio-Jimdo** [google/cpu_features#179]
While installing some other node dependencies, one of them attempts to build `cpu_features` using `node-gyp` and fails with the following message. Is there an environment variable the code will look for to control this, or is it simply an issue on `Apple M1` or am I jus totally off-base?

```sh
Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Users/gio/Code/devex-backstage/node_modules/cpu-features
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@14.17.6 | darwin | arm64
gyp info find Python using Python version 3.9.7 found at \"/Users/gio/.pyenv/versions/3.9.7/bin/python\"
gyp info spawn /Users/gio/.pyenv/versions/3.9.7/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/gio/Code/devex-backstage/node_modules/@npmcli/run-script/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/gio/Code/devex-backstage/node_modules/cpu-features/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/gio/Code/devex-backstage/node_modules/@npmcli/run-script/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/gio/Library/Caches/node-gyp/14.17.6/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/gio/Library/Caches/node-gyp/14.17.6',
gyp info spawn args   '-Dnode_gyp_dir=/Users/gio/Code/devex-backstage/node_modules/@npmcli/run-script/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/gio/Library/Caches/node-gyp/14.17.6/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/gio/Code/devex-backstage/node_modules/cpu-features',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  ACTION Configuring dependencies /Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build/Makefile
-- The C compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for getauxval
-- Looking for getauxval - not found
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build
  TOUCH Release/obj.target/config_deps.stamp
  ACTION Building dependencies /Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build/libcpu_features.a
gmake[1]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[2]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
[ 11%] Building C object CMakeFiles/utils.dir/src/filesystem.c.o
[ 22%] Building C object CMakeFiles/utils.dir/src/stack_line_reader.c.o
[ 33%] Building C object CMakeFiles/utils.dir/src/string_view.c.o
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
[ 33%] Built target utils
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
[ 44%] Building C object CMakeFiles/unix_based_hardware_detection.dir/src/hwcaps.c.o
[ 55%] Building C object CMakeFiles/unix_based_hardware_detection.dir/src/unix_features_aggregator.c.o
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
[ 55%] Built target unix_based_hardware_detection
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[3]: Entering directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
[ 66%] Building C object CMakeFiles/cpu_features.dir/src/cpuinfo_arm.c.o
In file included from /Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/src/cpuinfo_arm.c:15:
/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/include/cpuinfo_arm.h:118:2: error: \"Including cpuinfo_arm.h from a non-arm target.\"
#error \"Including cpuinfo_arm.h from a non-arm target.\"
 ^
1 error generated.
gmake[3]: *** [CMakeFiles/cpu_features.dir/build.make:76: CMakeFiles/cpu_features.dir/src/cpuinfo_arm.c.o] Error 1
gmake[3]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[2]: *** [CMakeFiles/Makefile2:90: CMakeFiles/cpu_features.dir/all] Error 2
gmake[2]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
gmake[1]: *** [Makefile:136: all] Error 2
gmake[1]: Leaving directory '/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build'
make: *** [/Users/gio/Code/devex-backstage/node_modules/cpu-features/deps/cpu_features/build/libcpu_features.a] Error 2
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/gio/Code/devex-backstage/node_modules/@npmcli/run-script/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Darwin 20.6.0
gyp ERR! command \"/Users/gio/.nvm/versions/node/v14.17.6/bin/node\" \"/Users/gio/Code/devex-backstage/node_modules/.bin/node-gyp\" \"rebuild\"
gyp ERR! cwd /Users/gio/Code/devex-backstage/node_modules/cpu-features
```

**Comment by Gio-Jimdo** [comment]
> seem related to #121
> 
> https://github.com/google/cpu_features/blob/628c50e92d7b2223a3159fb53dbc49835233a84a/include/cpuinfo_arm.h#L117-L119
> 
> https://github.com/google/cpu_features/blob/119943707c42f42a5f2344f0c0ac12bc2190b3d8/include/cpu_features_macros.h#L38-L40

... and how would one set an environment variable that un-sets these flags?

**Comment by Gio-Jimdo** [comment]
Is this issue addressed by PR #150?
